src/dackar/knowledge_graph/py2neo.py

```diff
@@ -0,0 +1,267 @@
+            session.execute_write(self._create_relation, l1, p1, l2, p2, lr, pr)
+
+    @staticmethod
+    def _create_relation(tx, l1, p1, l2, p2, lr, pr):
+        """create graph relation
+
+        Args:
+            tx (obj): python neo4j active session that can be used to execute queries
+            l1 (str): first node label
+            p1 (dict): first node attributes
+            l2 (str): second node label
+            p2 (dict): second node attributes
+            lr (str): relationship label
+            pr (dict, optional): attributes for relationship. Defaults to None.
+        """
+        if pr is not None:
+            query = f"""
+                MERGE (l1:{l1} {{ {', '.join([f'{k}:"{v}"' for k, v in p1.items()])} }})
+                MERGE (l2:{l2} {{ {', '.join([f'{k}:"{v}"' for k, v in p2.items()])} }})
+                MERGE (l1)-[r:{lr} {{ {', '.join([f'{k}: ${k}' for k in pr.keys()])} }} ]->(l2)
+            """
+            tx.run(query, **pr)
+        else:
+            query = f"""
+                MERGE (l1:{l1} {{ {', '.join([f'{k}:"{v}"' for k, v in p1.items()])} }})
+                MERGE (l2:{l2} {{ {', '.join([f'{k}:"{v}"' for k, v in p2.items()])} }})
+                MERGE (l1)-[r:{lr}]->(l2)
+            """
+            # print(query)
+            tx.run(query)
+
+    def find_nodes(self, label, properties=None):
+        """Find the node in neo4j graph database
+
+        Args:
+            label (str): node label
+            properties (dict, optional): node attributes. Defaults to None.
+
+        Returns:
+            list: list of nodes
+        """
+        with self.__driver.session() as session:
+            result = session.execute_read(self._find_nodes, label, properties)
+            return result
+
+    @staticmethod
+    def _find_nodes(tx, label, properties):
+        """Find the node in neo4j graph database
+
+        Args:
+            tx (obj): python neo4j active session that can be used to execute queries
+            label (str): node label
+            properties (dict, optional): node attributes. Defaults to None.
+
+        Returns:
+            list: list of nodes
+        """
+        if properties is None:
+            query = f"MATCH (n:{label}) RETURN n"
+        else:
+            query = f"""MATCH (n:{label} {{ {', '.join([f'{k}:"{v}"' for k, v in properties.items()])} }}) RETURN n"""
+        result = tx.run(query)
+        values = [record.values() for record in result]
+        return values
+
+    def load_csv_for_nodes(self, file_path, label, attribute):
+        """Load CSV file to create nodes
+
+        Args:
+            file_path (str): file path for CSV file, location is relative to 'dbms.directories.import' or 'server.directories.import' in neo4j.conf file
+            label (str): node label
+            attribute (dict): node attribute from the CSV column names
+        """
+        with self.__driver.session() as session:
```

Comment:
should we check the file exists?

Reply:
This is not easy to check at this point. We allow the user to set the neo4j  import folder path outside the module. The file path provided is relative path to the import folder. Right now, it will error out if the file path is not correct. I would say let's skip it for now.